### PR TITLE
Extend evaluate on AbstractArray, instead of Array

### PR DESCRIPTION
**Pull request by lbenet**
This allows for evaluations using `IntervalBoxes`.

### CODE REVIEW

File: src/evaluate.jl [hunk 1]
```diff
@@ -264,6 +264,10 @@ function evaluate(a::TaylorN{T}, vals::NTuple{N,S}) where
     return sum( sort!(suma, by=abs2) )
 end
 
+# This should allow to evaluate IntervalBoxes appropriately
+evaluate(a::TaylorN{T}, vals::AbstractArray) where
```
[the remark this answers is not in the document]
Comment (lbenet) [reply src/evaluate.jl]: Thanks a lot! Good catch! I'll try your suggestion.

**Comment by lbenet** [comment]
Thanks again for the suggestion.

**Comment by dpsanders** [comment]
Now that I think about it, it would be best to avoid the splat (`...`)  if possible, for efficiency.
Can you just treat the argument as an iterator with a known length in the original `evaluate` function?
i.e. start by checking the length of the argument and then suppose that you can index into it?

**Comment by lbenet** [comment]
Thanks for the suggestion. I think it is worth considering it, though I am not sure how.

The original evaluate function (for `TaylorN`, which is the case of interest in this PR) takes a `TaylorN` and a `NTuple`, and is implemented through a `for` of known length [here](https://github.com/JuliaDiff/TaylorSeries.jl/blob/c0d48591fd658aaf6cd471a0f431c9fbaa185181/src/evaluate.jl#L243). So my guess is that it is coded in a way you propose, but maybe in a primitive way. Some of the remaining methods do splat vectors, or treat chains of parameters as a proper ntuple by splatting, simply to call the original method. I am not sure if there is a high penalty on doing that.

**Comment by dpsanders** [comment]
To be honest I just have the general idea that "splatting is slow", but I guess it would require actual benchmarks to check that. Let's just leave it for now.

**Comment by lbenet** [comment]
Do you agree to go ahead and merge this?

**Pull request merged**